res/skins/QMLDemo/EqKnob.qml

```diff
@@ -0,0 +1,37 @@
+import Mixxx.Controls 0.1 as MixxxControls
+import QtQuick 2.15
+
+Item {
+    id: root
+
+    required property string channelGroup
+    property alias key: knob.key
+
+    width: 35
+    height: 35
+
+    MixxxControls.Knob {
+        id: knob
+
+        anchors.fill: parent
+        group: "[EqualizerRack1_" + channelGroup + "_Effect1]"
+        arc: true
+        arcRadius: 15
+        arcColor: "#858585"
+        arcWidth: 2
+
+        background: Image {
+            source: "../LateNight/palemoon/knobs/knob_bg_master.svg"
+            width: 35
+            height: 30
+        }
+
+        foreground: Image {
+            source: "../LateNight/palemoon/knobs/knob_indicator_regular_red.svg"
+            width: 35
+            height: 30
+        }
```

Comment:
You could use a Component to keep the code a bit more DRY:
```suggestion
        knobImage: Component {
            property alias source: image.source
            Image: {
              id: image
              width: 35
              height: 30
            }
        }
        background: knobImage {
            source: "../LateNight/palemoon/knobs/knob_bg_master.svg"
        }

        foreground: knobImage {
            source: "../LateNight/palemoon/knobs/knob_indicator_regular_red.svg"
        }
```
Its overkill in this case, but a handy technique to create components inline if they don't deserve their own file.

Reply:
Good point, thanks. Does this work in all QtQuick versions? I have Qt 5.15 and don't know if the current version even works on Qt 5.12.

Reply:
I'm pretty pretty sure it does work on all versions.
Making the alias properties required would make sense as well, but I'm not sure if that has implications for the Qt version.

Reply:
> Making the alias properties required would make sense as well

Nope, not possible. The normal properties you use one of these:

    [default] [required] [readonly] property <propertyType> <propertyName>
    [default] property <propertyType> <propertyName> : <value>

But for alias properties the signature is:

  [default] property alias <name>: <alias reference>

So you can't make them required.

https://doc.qt.io/qt-5/qtqml-syntax-objectattributes.html#property-aliases

Reply:
Ok, good to know, so I guess alias properties would get required automatically when they refer to a required resource?

Reply:
I guess so.